homeassistant/components/energyzero/__init__.py

```diff
@@ -1,17 +1,86 @@
 """The EnergyZero integration."""
 from __future__ import annotations
 
+from datetime import date, datetime
+
+from energyzero import Electricity, Gas
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
-from homeassistant.core import HomeAssistant
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
 from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.util import dt as dt_util
 
-from .const import DOMAIN
+from .const import (
+    ATTR_END,
+    ATTR_INCL_VAT,
+    ATTR_START,
+    DOMAIN,
+    ENERGY_SERVICE_NAME,
+    GAS_SERVICE_NAME,
+    SERVICE_SCHEMA,
+    PriceType,
+)
 from .coordinator import EnergyZeroDataUpdateCoordinator
 
 PLATFORMS = [Platform.SENSOR]
 
 
+def __get_date(date_input: str | None) -> date | datetime:
+    """Get date."""
+    if not date_input:
+        return dt_util.now().date()
+
+    if value := dt_util.parse_datetime(date_input):
+        return value
+
+    raise ValueError(f"Invalid date: {date_input}")
+
+
+def __serialize_prices(prices: Electricity | Gas) -> ServiceResponse:
+    """Serialize prices."""
+    return {str(timestamp): price for timestamp, price in prices.prices.items()}
+
+
+async def __get_prices(
+    coordinator: EnergyZeroDataUpdateCoordinator,
+    call: ServiceCall,
+    price_type: PriceType,
+) -> ServiceResponse:
+    previous_incl_vat = coordinator.energyzero.incl_vat
+
+    try:
+        coordinator.energyzero.incl_vat = str(call.data[ATTR_INCL_VAT]).lower()
+        start = __get_date(call.data.get(ATTR_START))
+        end = __get_date(call.data.get(ATTR_END))
+
+        data: Electricity | Gas
+
+        if price_type == PriceType.GAS:
+            data = await coordinator.energyzero.gas_prices(
```

Comment:
I think it would be great to extend the third-party library to allow passing the `incl_vat` as setting and restoring is hacking and there could be some race conditions

Reply:
@klaasnicolaas is this the correct way to solve this issue: https://github.com/klaasnicolaas/python-energyzero/pull/270

Reply:
Yes, but superseded by a PR of mine 😅 I will create a separate PR for the dependency bump.

Reply:
If it is allowed, I can also just include it here as I need the updated version anyway?

Reply:
You can update the dependency locally, just don't commit it. Once the PR (#105080) has been merged, you only need to rebase your branch so that you are up to date with all changes from dev.